### PR TITLE
GEODE-7294: Update dependencies for v1.12

### PR DESCRIPTION
**Pull request by metatype**
Update many but not all dependencies.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-lucene/src/integrationTest/java/org/apache/geode/cache/lucene/LuceneIndexCreationPersistenceIntegrationTest.java [hunk 1]
```diff
@@ -228,7 +228,7 @@ private void verifyQueryResultSize(String indexName, String regionName, String q
       String defaultField, int size) throws Exception {
     LuceneQuery query = luceneService.createLuceneQueryFactory().create(indexName, regionName,
         queryString, defaultField);
-    await().untilAsserted(() -> assertThat(query.findPages()).hasSize(size));
+    await().untilAsserted(() -> assertThat(query.findPages()).toIterable().hasSize(size));
```
Comment (upthewaterspout) [inline geode-lucene/src/integrationTest/java/org/apache/geode/cache/lucene/LuceneIndexCreationPersistenceIntegrationTest.java]: odd that this needed to change for a minor upgrade.

**Comment by gesterzhou** [comment]
I need to run lucene backward compatibility tests

File: geode-lucene/src/integrationTest/java/org/apache/geode/cache/lucene/LuceneIndexCreationPersistenceIntegrationTest.java (hunk 1 above)
Comment (metatype) [reply geode-lucene/src/integrationTest/java/org/apache/geode/cache/lucene/LuceneIndexCreationPersistenceIntegrationTest.java]: This was changed with assertj 3.12.0.  See https://assertj.github.io/doc/#assertj-core-3-12-0-release-notes.

**Comment by gesterzhou** [comment]
The  lucene backward compatibility regression passed. Precheckin also passed.

**Pull request merged**